project/src/main/java/com/google/sps/servlets/CreateProductServlet.java

```diff
@@ -138,13 +141,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
     
     
-    // TODO:Phillips Product Search functions 
+  }
+
+  public void createAndAddToProductSearch(String productId, String productSetId, String productDisplayName, String productCategory, HttpServletRequest request) throws IOException{ 
     // Functions to create product and add to a product set in the product search database
     ProductSearchLibrary.createProduct(productId, productDisplayName, productCategory);
+
     ProductSearchLibrary.addProductToProductSet(productId, productSetId);
-    
+
     //Create reference image for a product to facilitate the searching for a product by image
     //image gcsuri used for reference image id
-    ProductSearchLibrary.createReferenceImage(productId, request.getParameter("mainGcsUrl"), request.getParameter("mainGcsUrl"));
+    String gcsUri = request.getParameter("mainGcsUrl");
+    String objectName = gcsUri.substring(gcsUri.lastIndexOf('/') + 1);
+
+    gcsUri = changeGcsFormat(gcsUri);
+    
+    ProductSearchLibrary.createReferenceImage(productId, objectName, gcsUri);
+    
+  }
+
+  public String changeGcsFormat(String gcsUri){
```

Comment:
This can probably be `private` since I don't think you'll use it elsewhere (same for createAndAddToProductSearch?)